internal/controller/phases.go

```diff
@@ -562,6 +572,65 @@ func getProvider(provider operatorv1.GenericProvider, defaultVersion string) clu
 	return *clusterctlProvider
 }
 
+func (p *phaseReconciler) getCustomProviders(ctx context.Context) ([]operatorv1.GenericProvider, error) {
```

Comment:
Also, please move this function to the `util` package. You will need to utilize `genericProviderList` for casting which is not exporting.